ibtk/src/lagrangian/FEDataManager.cpp

```diff
@@ -871,10 +871,12 @@ FEDataManager::spread(const int f_data_idx,
                 {
                     for (unsigned int qp = 0; qp < n_qp; ++qp)
                     {
-                        for (unsigned int i = 0; i < n_vars; ++i)
-                        {
-                            F_JxW_qp[n_vars * (qp_offset + qp) + i] += F_node[k][i] * phi_F[k][qp] * JxW_F[qp];
-                        }
+                        const int idx = n_vars * (qp_offset + qp);
+                        const double p_JxW_F = phi_F[k][qp] * JxW_F[qp];
+                        if (n_vars == NDIM)
```

Comment:
Is this simply to make it easier for the compiler to vectorize, so that it knows the size of the loop?

Another common case would probably be `n_vars == 1`.

Reply:
More-or-less yes. I didn't look at the assembly; I don't know if it would vectorize but it will surely unroll this loop.

Reply:
Regarding `n_vars == 1`: when does that happen in this function? I thought that we always had an NDIM-dimensional force.

Reply:
You can "spread" quantities that are not forces.

Reply:
(It can make sense to spread any quantity that is a density.)

Reply:
I thought about this some more and figured that the best solution would be to simply add a second set of template functions for computing this. I added a test to #494 that verifies that we still calculate the same values (up to roundoff) for at least one spread kernel.